lib/puppet/configurer.rb

```diff
@@ -502,8 +513,8 @@ def retrieve_new_catalog(query_options)
         Puppet[:node_name_value],
         query_options.merge(
           :ignore_cache      => true,
-          # We never want to update the cached Catalog if we're running in noop mode.
-          :ignore_cache_save => Puppet[:noop],
+          # don't update cache until after environment converges
+          :ignore_cache_save => true,
```

Comment:
Are we effectively making this option irrelevant now?

Reply:
We unfortunately still need to be able to support the case `puppet agent --noop --use_cached_catalog` where we load the catalog from the cache, but don't update the cache. Once we switch the configurer over to using the http client to retrieve the catalog, then we can remove all of this.